homeassistant/components/wilight/light.py

```diff
@@ -0,0 +1,223 @@
+"""Support for WiLight lights."""
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_HS_COLOR,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR,
+    LightEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from . import WiLightDevice
+from .const import (
+    DOMAIN,
+    ITEM_LIGHT,
+    LIGHT_COLOR,
+    LIGHT_DIMMER,
+    LIGHT_NONE,
+    LIGHT_ON_OFF,
+    SUPPORT_NONE,
+)
+
+
+def entities_from_discovered_wilight(hass, api_device):
+    """Parse configuration and add WiLight light entities."""
+    entities = []
+    for item in api_device.items:
+        if item["type"] != ITEM_LIGHT:
+            continue
+        if item["sub_type"] == LIGHT_NONE:
+            continue
+        index = item["index"]
+        item_name = item["name"]
+        if item["sub_type"] == LIGHT_ON_OFF:
+            entity = WiLightLightOnOff(api_device, index, item_name)
+        elif item["sub_type"] == LIGHT_DIMMER:
+            entity = WiLightLightDimmer(api_device, index, item_name)
+        elif item["sub_type"] == LIGHT_COLOR:
+            entity = WiLightLightColor(api_device, index, item_name)
+        else:
+            continue
+        entities.append(entity)
+
+    return entities
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+):
+    """Set up WiLight lights from a config entry."""
+    parent = hass.data[DOMAIN][entry.entry_id]
+
+    # Handle a discovered WiLight device.
+    entities = entities_from_discovered_wilight(hass, parent.api)
+    async_add_entities(entities)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+
+    return True
+
+
+class WiLightLightOnOff(WiLightDevice, LightEntity):
+    """Representation of a WiLights light on-off."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the device."""
+        WiLightDevice.__init__(self, *args, **kwargs)
+        # Initialize the WiLights onoff.
+        self._on = False
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_NONE
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        if "on" in self._status:
+            self._on = self._status["on"]
+        return self._on
```

Comment:
Why do we need the instance attribute?
```suggestion
        return self._status.get("on")
```

Reply:
At startup, self._status = {}.
self._status is updated with IO and may not contain "on", "brightness", "hue" or "saturation" at the beginning.

That is why:

    def __init __ (self, * args, ** kwargs):
        "" "Initialize the device." ""
        WiLightDevice .__ init __ (self, * args, ** kwargs)
        # Initialize the WiLights color.
        self._on = False
        self._brightness = 0
        self._hue = 0
        self._saturation = 100

    @property
    def brightness (self):
        "" "Return the brightness of this light between 0..255." ""
        if "brightness" in self._status:
            self._brightness = int (self._status ["brightness"])
        return self._brightness

    @property
    def hs_color (self):
        "" "Return the hue and saturation color value [float, float]." ""
        if "hue" in self._status:
            self._hue = wilight_to_hass_hue (int (self._status ["hue"]))
        if "saturation" in self._status:
            self._saturation = wilight_to_hass_saturation (
                int (self._status ["saturation"])
            )
        return [self._hue, self._saturation]

    @property
    def is_on (self):
        "" "Return true if device is on." ""
        if "on" in self._status:
            self._on = self._status ["on"]
        return self._on

Reply:
Using `dict.get` as in my suggestion handles that case.

Reply:
OK, Thanks